plat/nvidia/tegra/common/tegra_sip_calls.c

```diff
@@ -85,6 +95,24 @@ uint64_t tegra_sip_handler(uint32_t smc_fid,
 
 		SMC_RET1(handle, 0);
 
+	case TEGRA_SIP_AARCH_SWITCH:
```

Comment:
This is a SMC32 call - based on the smc_fid value. Thus you should clean the high bits of the x1/x2 parameters
`x1 = (uint32_t)x1;`
etc. before using them. In the case that a call came from AArch32, it is not guaranteed that the high bits are zero, which this code appears to depend on.

I notice that this also affects the VIDEOMEM_REGION function, but I can't add a comment there in this commit diff.

 I presume you already realise that these two SIP service calls cannot be passed addresses above 4GB as they are 32-bit functions.